### PR TITLE
Feature/53892 lead list back end

### PR DESCRIPTION
**Pull request by douglasalizares**
Se implementó los end points con tests para la lista de leads

### CODE REVIEW

**Review by ebrjose — APPROVED** [review]
Looks good..

File: src/controllers/lead/leadDtoBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export class LeadDtoBuilder {
+  leads: Lead[];
+
+  constructor(leads: Lead[]) {
+    this.leads = leads;
+  }
+
+  getLeadDto(): LeadDTO[] {
+    let index = -1;
+    return this.leads.map(() => {
```
Comment (marcelo-yourday) [inline src/controllers/lead/leadDtoBuilder.ts:13]: I think an easier way to do it is:

```javascript
return this.leads.map((lead) => ({
  contactName: lead.contactName, 
  email: lead.email,
  phone: lead.phone
});
```

File: src/data/iLeadRepository.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { Lead } from '../dto/leadDtoJsonFile';
+
+export interface ILeadRepository {
+  getAllLeads(): Lead[];
```
Comment (marcelo-yourday) [inline src/data/iLeadRepository.ts:4]: You can use a better abstraction for a Repository:

```javascript
interface IRepository<ModelType> {
  getAll(): ModelType[];
}
```

File: src/dto/leadDtoJsonFile.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export interface Lead {
```
Comment (marcelo-yourday) [inline src/dto/leadDtoJsonFile.ts:1]: I'm not sure if the name of this file is accurate, it seems we are defining a Lead, so, better if the file is called: lead.ts

**Review by marcelo-yourday — CHANGES_REQUESTED** [review]

File: src/controllers/lead/leadDtoBuilder.ts (hunk 1 above)
Comment (douglasalizares) [reply src/controllers/lead/leadDtoBuilder.ts]: solved

File: src/data/iLeadRepository.ts (hunk 2 above)
Comment (douglasalizares) [reply src/data/iLeadRepository.ts]: resolved

File: src/dto/leadDtoJsonFile.ts (hunk 3 above)
Comment (douglasalizares) [reply src/dto/leadDtoJsonFile.ts]: Resolved

**Review by marcelo-yourday — APPROVED** [review]

**Pull request merged**